enterprise/server/remote_execution/snaploader/snaploader.go

```diff
@@ -42,21 +42,33 @@ const (
 	chunkedFileWriteConcurrency = 4
 )
 
-// NewKey returns the cache key for a snapshot.
+// SnapshotKeys returns the cache keys for potential snapshot matches,
+// as well as the key that should be written to.
+//
 // TODO: include a version number in the key somehow, so that
 // if we make breaking changes e.g. to the vmexec API or firecracker
 // version etc., we can ensure that incompatible snapshots don't get reused.
-func NewKey(task *repb.ExecutionTask, configurationHash, runnerID string) (*fcpb.SnapshotKey, error) {
-	pd, err := digest.ComputeForMessage(task.GetCommand().GetPlatform(), repb.DigestFunction_SHA256)
+func SnapshotKeys(task *repb.ScheduledTask, configurationHash, runnerID string) (*fcpb.SnapshotKeys, error) {
+	pd, err := digest.ComputeForMessage(task.GetExecutionTask().GetCommand().GetPlatform(), repb.DigestFunction_SHA256)
```

Comment:
kind of a nit, but when I see lots of deep references like this (especially repeated ones), I'll often pull define the common bit like:

```
func SnapshotKeys(scheduled *repb.ScheduledTask, ...

task := scheduled.GetExecutionTask()
```

then you can shorten this to `task.GetCommand().GetPlatform()` and below to `task.GetExecuteRequest().GetInstanceName()`

Reply:
done